lib/pinyin/zhuyin_parsers.ex

```diff
@@ -0,0 +1,118 @@
+defmodule Pinyin.ZhuYinParsers do
+
+  import NimbleParsec
+  alias Pinyin.Parsers.Utils, as: Utils
+
+  # Empty tone must be last for parsing to work
+  zhuyin_tones = ["˙", "ˊ", "ˇ", "ˋ", ""]
+
+  initials = ~w(
+    ㄅ ㄆ ㄇ ㄈ
+    ㄉ ㄊ ㄋ ㄌ
+    ㄍ ㄎ ㄏ
+    ㄐ ㄑ ㄒ
+  )
+
+  standalone_initials = ~w(
+    ㄓ ㄔ ㄕ ㄖ
+    ㄗ ㄘ ㄙ
+  )
+
+  two_finals = ~w(
+    ㄧㄚ
+    ㄨㄚ
+    ㄧㄥ
+    ㄧㄤ
+    ㄧㄝ
+    ㄨㄛ
+    ㄨㄥ
+    ㄨㄤ
+    ㄧㄠ
+    ㄨㄞ
+    ㄩㄝ
+    ㄩㄥ
+    ㄧㄡ
+    ㄨㄟ
+    ㄩㄝ
+    ㄧㄢ
+    ㄨㄢ
+    ㄩㄢ
+    ㄧㄣ
+    ㄨㄣ
+    ㄩㄣ
+  )
+  single_finals = ~w(
+    ㄧ ㄨ ㄩ
+    ㄚ ㄛ ㄜ ㄦ
+    ㄞ ㄟ ㄠ ㄡ
+    ㄢ ㄣ ㄤ ㄥ
+  )
+
+  initials_parser =
+    Enum.concat(initials, standalone_initials) |> Utils.wordlist_to_parser
+
+  standalone_initials_parser =
+    standalone_initials |> Utils.wordlist_to_parser
+
+  finals_parser =
+    choice([
+      two_finals |> Utils.wordlist_to_parser,
+      single_finals |> Utils.wordlist_to_parser
+    ])
+
+  tone_parser =
+    zhuyin_tones |> Utils.wordlist_to_parser
+
+  defp to_zhuyin([initial, final, tone]) do
+    Zhuyin.create(initial, final, tone)
+  end
+  defp to_zhuyin([final, tone]) do
+    Zhuyin.create(final, tone)
+  end
+
+  zhuyin_word =
+    choice([
+      finals_parser,
+      concat(initials_parser, finals_parser),
+      standalone_initials_parser,
+    ])
+    |> concat(optional(tone_parser))
+    |> reduce({:to_zhuyin, []})
+
+  defparsec(
+    :zhuyin_word,
+    Utils.repeat_choice_parser([
+      zhuyin_word,
+      Utils.split(),
+      Utils.not_split()
+    ]),
+    inline: true
+  )
+
+  defparsec(
+    :zhuyin_only,
+    Utils.repeat_choice_parser([zhuyin_word, Utils.split()]),
+    inline: true
+  )
+
+  defparsec(
+    :zhuyin_words,
+    Utils.repeat_choice_parser([
+      zhuyin_word,
+      Utils.split(),
+      Utils.not_split()
+    ]),
+    inline: true
+  )
+
+  # TODO: Do I need syllable?
```

Comment:
TODO: Pinyin uses `pinyin_syllable.` Check what that is exactly and whether I need to do the same here.

Reply:
It's been some time since I wrote the Pinyin parser, but I _think_ the syllable vs word distinction refers to writing "nihao" vs "ni hao". The former is one word containing two syllables, while the latter are two words containing two syllables each. Not sure which spacing is used for Zhuyin though, so I'll leave the choice up to you :).

Reply:
Ah I see. Need the same for zhuyin.